src/quantuminspire/projectq/backend_qx.py

```diff
@@ -645,19 +721,27 @@ def __init__(self, qubit_id: int) -> None:
 
         random_measurement = self._sample_measured_states_once()
 
-        for logical_qubit_id in self._measured_ids:
-            physical_qubit_id = self._logical_to_physical(logical_qubit_id)
-            sim_qubit_id = self._physical_to_simulated(physical_qubit_id)
+        if self._full_state_projection:
```

Comment:
optional: this if-else block is identical to the block in [672-678](https://github.com/QuTech-Delft/quantuminspire/pull/140/files#diff-9c941475897669e0a7260eceed76b5f53188f6f66e8632068dd885934070c23aR672-R678), with the exception of the `measurement_block_index`. Is it worth extracting into a separate function?